### PR TITLE
CIRC-1360 Update logic for creating recall tlr requests

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Update logic for fetching loans and items for recall requests

## Approach
When trying to create a title-level request, and multiple open loans exist for items of the requested instance, we should recall the loan with the closest due date. In such a case, an item should be linked to a title-level request (itemId and holdingsRecordId fields should be filled)

Resolves: [CIRC-1360](https://issues.folio.org/browse/CIRC-1360)

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java [hunk 1]
```diff
@@ -259,19 +259,20 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
     }
 
     JsonObject currentItemRequest = itemRequests.get(startIndex);
-
+    ItemRepository itemRepository = new ItemRepository(clients, true, false, false);
```
Comment (sherzodn) [inline src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java:262]: using multiple boolean arguments does not look smell, would prefer enums

Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java:262]: what do you mean by preferring enums?

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java [hunk 2]
```diff
@@ -150,13 +153,15 @@ void replace(RoutingContext routingContext) {
       updateRequestQueue, new ClosedRequestValidator(requestRepository),
       requestNoticeSender, updateItem, eventPublisher);
 
+    final var itemRepository = new ItemRepository(clients, true, true, true);
     final var requestFromRepresentationService = new RequestFromRepresentationService(
       new InstanceRepository(clients),
       new ItemRepository(clients, true, true, true),
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/RequestCollectionResource.java:159]: replace with itemRepository

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 3]
```diff
@@ -2583,6 +2584,73 @@ void cannotCreateRequestWithItemIdButNoHoldingsRecordId(RequestLevel requestLeve
...
+      .withRequestDate(requestDate)
+      .fulfilToHoldShelf()
+      .withRequestExpiration(LocalDate.of(2021, 7, 30))
+      .withHoldShelfExpiration(LocalDate.of(2021, 8, 31))
+      .withPickupServicePointId(pickupServicePointId)
+      .withTags(new RequestBuilder.Tags(asList("new", "important")))
+      .withPatronComments("I need this book"));
+
+    var recalledLoanJson = loansFixture.getLoanById(firstLoan.getId()).getJson();
+    assertThat(recalledLoanJson.getBoolean("dueDateChangedByRecall"), is(true));
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:2620]: let's check due date

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 4]
```diff
@@ -2583,6 +2584,73 @@ void cannotCreateRequestWithItemIdButNoHoldingsRecordId(RequestLevel requestLeve
...
+      .by(requester)
+      .withRequestDate(requestDate)
+      .fulfilToHoldShelf()
+      .withRequestExpiration(LocalDate.of(2021, 7, 30))
+      .withHoldShelfExpiration(LocalDate.of(2021, 8, 31))
+      .withPickupServicePointId(pickupServicePointId)
+      .withTags(new RequestBuilder.Tags(asList("new", "important")))
+      .withPatronComments("I need this book"));
+
+    var recalledLoanJson = loansFixture.getLoanById(firstLoan.getId()).getJson();
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:2619]: Let's check second loan

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**